zookeeper-server/src/test/java/org/apache/zookeeper/common/ZKTestClientSSLContext.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.zookeeper.common;
+
+import javax.net.ssl.SSLContext;
+
+public class ZKTestClientSSLContext implements ZKClientSSLContext {
```

Comment:
I think this can be an inner class in X509UtilTest since it's not used anywhere else

Reply:
Can do, though moving it to an inner class would require it to be static, and the property would have to be something like: 

`zkConfig.setProperty(clientX509Util.getSslContextSupplierClassProperty(), X509UtilTest.class.getCanonicalName() + "$" + TestContextSupplier.class.getSimpleName());`

Reply:
Ok, if instantiating an inner class by reflection is awkward, don't worry about it and leave it as a standalone class.

Reply:
Another nit then - can you make the class package-private? Principle of least privilege and all that (Not sure if a package-private class can be instantiated using reflection though).

Reply:
There's an argument for both ways - I've a feeling that package-private might mess with reflection, but I'll test it out. The static inner class works fine, just the property looks a bit funny - but that isn't too important for the test case.